src/systems/environment_preload/VisualizationTool.cc

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "VisualizationTool.hh"
+
+/////////////////////////////////////////////////
+EnvironmentVisualizationTool::EnvironmentVisualizationTool()
+{
+    this->pcPub =
+      this->node.Advertise<gz::msgs::PointCloudPacked>("/point_cloud");
+}
+
+/////////////////////////////////////////////////
+void EnvironmentVisualizationTool::CreatePointCloudTopics(
+    const std::shared_ptr<components::EnvironmentalData> _data) {
+  this->pubs.clear();
+  this->sessions.clear();
+
+  for (auto key : _data->frame.Keys())
+  {
+    this->pubs.emplace(key, node.Advertise<gz::msgs::Float_V>(key));
+    gz::msgs::Float_V msg;
+    this->floatFields.emplace(key, msg);
+    this->sessions.emplace(key, _data->frame[key].CreateSession());
+  }
+}
+
+/////////////////////////////////////////////////
+void EnvironmentVisualizationTool::FileReloaded()
+{
+  this->finishedTime = false;
+}
+
+/////////////////////////////////////////////////
+void EnvironmentVisualizationTool::Step(
+    const UpdateInfo &_info,
+    const EntityComponentManager& _ecm,
+    const std::shared_ptr<components::EnvironmentalData> _data,
+    double _xSamples, double _ySamples, double _zSamples)
+{
+
+  auto now = std::chrono::steady_clock::now();
+  std::chrono::duration<double> dt(now - this->lastTick);
+
+  if (this->resample)
+  {
+    this->CreatePointCloudTopics(_data);
+    this->ResizeCloud(_data, _ecm, _xSamples, _ySamples, _zSamples);
+    this->resample = false;
+    this->lastTick = now;
+  }
+
+  for (auto &it : this->sessions)
```

Comment:
Can we have a brief comment here saying this for-loop is stepping the timestamps until it runs out (I assume that's what it's doing), so that the reader doesn't have to read the code to know that's what it's doing?